include/color_spinor_field_order.h

```diff
@@ -951,20 +972,19 @@ namespace quda {
       const AllocInt norm_offset;
       int volumeCB;
       int faceVolumeCB[4];
-      int stride;
       mutable Float *ghost[8];
       mutable norm_type *ghost_norm[8];
       int nParity;
       void *backup_h; //! host memory for backing up the field when tuning
       size_t bytes;
 
-      FloatNOrder(const ColorSpinorField &a, int nFace = 1, Float *field_ = 0, norm_type *norm_ = 0, Float **ghost_ = 0) :
-        field(field_ ? field_ : (Float *)a.V()),
-        norm(norm_ ? norm_ : (norm_type *)a.Norm()),
+      FloatNOrder(const ColorSpinorField &a, int nFace = 1, Float *buffer = 0, Float **ghost_ = 0) :
+        field(buffer ? buffer : (Float *)a.V()),
+        norm(buffer ? reinterpret_cast<norm_type *>(reinterpret_cast<char *>(buffer) + a.NormOffset()) :
+                      const_cast<norm_type *>(reinterpret_cast<const norm_type *>(a.Norm()))),
         offset(a.Bytes() / (2 * sizeof(Float) * N)),
-        norm_offset(a.NormBytes() / (2 * sizeof(norm_type))),
+        norm_offset(a.Bytes() / (2 * sizeof(norm_type))),
```

Comment:
Similar to above, should this be divided by `2 * sizeof(Float))` instead of `2 * sizeof(norm_type)`?

Reply:
Nevermind, misunderstood the purpose of the variable. All good here.